### ISSUE TITLE
[Bug]: task='conversational' not working

### ISSUE THREAD

**Issue by duc-phamh** [BerriAI/litellm#392]
### What happened?

Here is my code when testing out the new `task='conversational'` feature:

```
messages = [
    {
        "content": "Hello! How are you today?",
        "role": "user"
    },
]

response = litellm.completion(
    model='huggingface/facebook/blenderbot-400M-distill', 
    messages=messages,
    api_key='api_key',
    api_base='api_base',
    stream=False,
    task='conversational'
)

print(response['choices'][0]['message']['content'])

```

Please see the log below for more details. `return_full_text` is a parameter used in the `text_generation_inference` mode. Somehow it is being passed into the `conversational` mode.

### Relevant log output

```shell
HuggingfaceError                          Traceback (most recent call last)
File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/main.py:591, in completion(model, messages, functions, function_call, temperature, top_p, n, stream, stop, max_tokens, presence_penalty, frequency_penalty, logit_bias, user, deployment_id, return_async, mock_response, api_key, api_version, api_base, force_timeout, num_beams, logger_fn, verbose, azure, custom_llm_provider, litellm_call_id, litellm_logging_obj, use_client, id, top_k, task, return_full_text, remove_input, request_timeout, fallbacks, caching, cache_params)
    584 huggingface_key = (
    585     api_key
    586     or litellm.huggingface_key
   (...)
    589     or litellm.api_key
    590 )
--> 591 model_response = huggingface_restapi.completion(
    592     model=model,
    593     messages=messages,
    594     api_base=api_base, # type: ignore
    595     model_response=model_response,
    596     print_verbose=print_verbose,
    597     optional_params=optional_params,
    598     litellm_params=litellm_params,
    599     logger_fn=logger_fn,
    600     encoding=encoding, 
    601     api_key=huggingface_key, 
    602     logging_obj=logging,
    603     custom_prompt_dict=litellm.custom_prompt_dict
    604 )
    605 if "stream" in optional_params and optional_params["stream"] == True:
    606     # don't try to access stream object,

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/llms/huggingface_restapi.py:144, in completion(model, messages, api_base, model_response, print_verbose, encoding, api_key, logging_obj, custom_prompt_dict, optional_params, litellm_params, logger_fn)
    143     print_verbose(f"response.status_code: {response.status_code}")
--> 144     raise HuggingfaceError(
    145         message=completion_response["error"],
    146         status_code=response.status_code,
    147     )
    148 else:

HuggingfaceError: The following `model_kwargs` are not used by the model: ['return_full_text'] (note: typos in the generate arguments will also show up in this list)

During handling of the above exception, another exception occurred:

InvalidRequestError                       Traceback (most recent call last)
Cell In[5], line 8
      1 messages = [
      2     {
      3         "content": "Hello! How are you today?",
      4         "role": "user"
      5     },
      6 ]
----> 8 response = litellm.completion(model='huggingface/facebook/blenderbot-400M-distill', messages=messages,
      9     api_key='dhkuuFbeULziDTWghuvppGtvCkTUZewYTsumVVROCqyludeVKqrxoVyLHqrjLojYRMGsIZoZWlzKoPaEDarDrwPWZNQoQRjdhJRPggxJncKOUBJSvNtjEENDxkPVezcg',
     10     api_base='https://wvrxeejles4fdyo5.us-east-1.aws.endpoints.huggingface.cloud',
     11     # max_tokens=1000,
     12     stream=False,
     13     task='conversational'
     14 )
     16 # for chunk in response:
     17 #     content = chunk["choices"][0].get("delta", {}).get("content")
     18 #     print(content, end='')
     19 print(response['choices'][0]['message']['content'])

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/utils.py:638, in client..wrapper(*args, **kwargs)
    634     if (
    635         liteDebuggerClient and liteDebuggerClient.dashboard_url != None
    636     ):  # make it easy to get to the debugger logs if you've initialized it
    637         e.message += f"\n Check the log in your dashboard - {liteDebuggerClient.dashboard_url}"
--> 638 raise e

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/utils.py:598, in client..wrapper(*args, **kwargs)
    595             return cached_result
    597 # MODEL CALL
--> 598 result = original_function(*args, **kwargs)
    599 end_time = datetime.datetime.now()
    600 if "stream" in kwargs and kwargs["stream"] == True:
    601     # TODO: Add to cache for streaming

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/timeout.py:44, in timeout..decorator..wrapper(*args, **kwargs)
     42     local_timeout_duration = kwargs["request_timeout"]
     43 try:
---> 44     result = future.result(timeout=local_timeout_duration)
     45 except futures.TimeoutError:
     46     thread.stop_loop()

File ~/miniforge3/envs/skydeck/lib/python3.11/concurrent/futures/_base.py:456, in Future.result(self, timeout)
    454     raise CancelledError()
    455 elif self._state == FINISHED:
--> 456     return self.__get_result()
    457 else:
    458     raise TimeoutError()

File ~/miniforge3/envs/skydeck/lib/python3.11/concurrent/futures/_base.py:401, in Future.__get_result(self)
    399 if self._exception:
    400     try:
--> 401         raise self._exception
    402     finally:
    403         # Break a reference cycle with the exception in self._exception
    404         self = None

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/timeout.py:33, in timeout..decorator..wrapper..async_func()
     32 async def async_func():
---> 33     return func(*args, **kwargs)

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/main.py:948, in completion(model, messages, functions, function_call, temperature, top_p, n, stream, stop, max_tokens, presence_penalty, frequency_penalty, logit_bias, user, deployment_id, return_async, mock_response, api_key, api_version, api_base, force_timeout, num_beams, logger_fn, verbose, azure, custom_llm_provider, litellm_call_id, litellm_logging_obj, use_client, id, top_k, task, return_full_text, remove_input, request_timeout, fallbacks, caching, cache_params)
    945     return response
    946 except Exception as e:
    947     ## Map to OpenAI Exception
--> 948     raise exception_type(
    949         model=model, custom_llm_provider=custom_llm_provider, original_exception=e
    950     )

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/utils.py:2284, in exception_type(model, original_exception, custom_llm_provider)
   2282 # don't let an error with mapping interrupt the user from receiving an error from the llm api calls
   2283 if exception_mapping_worked:
-> 2284     raise e
   2285 else:
   2286     raise original_exception

File ~/miniforge3/envs/skydeck/lib/python3.11/site-packages/litellm/utils.py:2018, in exception_type(model, original_exception, custom_llm_provider)
   2016 elif original_exception.status_code == 400:
   2017     exception_mapping_worked = True
-> 2018     raise InvalidRequestError(
   2019         message=f"HuggingfaceException - {original_exception.message}",
   2020         model=model,
   2021         llm_provider="huggingface",
   2022     )
   2023 elif original_exception.status_code == 408:
   2024     exception_mapping_worked = True

InvalidRequestError: HuggingfaceException - The following `model_kwargs` are not used by the model: ['return_full_text'] (note: typos in the generate arguments will also show up in this list)
```

**Comment by ghost** [comment]
@duc-phamh acknowleding this issue, i'll take care of this today.

**Comment by ghost** [comment]
@duc-phamh should be resolved from v`0.1.692` onwards

**Comment by duc-ph** [comment]
Thank you @krrishdholakia, this is working as expected 🙏
